### ISSUE TITLE
ESLint's cache can stop autofixing

### ISSUE THREAD

**Issue by MayhemYDG** [eslint/eslint#10679]
**Tell us about your environment**

* **ESLint Version:** 5.2.0
* **Node Version:** v10.7.0
* **npm Version:** 6.1.0

**What parser (default, Babel-ESLint, etc.) are you using?**
default or babel-eslint
**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```json
{
  "rules": {
    "quotes": 1
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
'test'
```

```bash
.\node_modules\.bin\eslint --cache src
.\node_modules\.bin\eslint --fix --cache src
```

**What did you expect to happen?**
ESLint should fix the fixable issues.

**What actually happened? Please include the actual, raw output from ESLint.**
```
C:\test> .\node_modules\.bin\eslint --cache src

C:\test\src\test.js
  1:1  warning  Strings must use doublequote  quotes

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.

C:\test> .\node_modules\.bin\eslint --fix --cache src

C:\test\src\test.js
  1:1  warning  Strings must use doublequote  quotes

✖ 1 problem (0 errors, 1 warning)
  0 errors and 1 warning potentially fixable with the `--fix` option.
```
ESLint does not autofix when the cache is populated. If I delete `.eslintcache`, then ESLint will autofix just fine.
I believe this is a regression since 5.2.0 (81283d0).
ESLint, when autofixing, shouldn't ignore files that are cached, do not invalidate the cache, but are known to have issues.

**Comment by MayhemYDG** [comment]
I should add: I like to verify a rule's configuration before running eslint with `--fix`, but I also want to make use of the cache so eslint doesn't have to re-lint files that have not changed and have no issues.

**Comment by MayhemYDG** [comment]
@platinumazure I think there might be a little misunderstanding: during the `--fix` run, the files are not fixed, they're just ignored.

**Comment by platinumazure** [comment]
Oh, thanks for clarifying.

I think I know what's going on. The cache info does not contain fix information because it was written when ESLint was run without autofix. When ESLint is run with autofix, it gets the cached lint results which don't have any autofix information, so ESLint decides there are no fixes to apply.

I think the best approach here will be to read from the cache, and if there are errors/warnings and autofix is enabled, we need to ignore the cache results and let the file be linted normally.

**Comment by platinumazure** [comment]
@MayhemYDG I've got a branch up in this repo: "fix-cache" (also represented in PR #10694). If you have a chance, I'd be grateful if you could take a look and see if it might fix this issue. I *think* I've figured it out, but an extra set of eyes never hurt. :smile:
